third_party/amd/lib/TritonAMDGPUTransforms/ReorderInstructions.cpp

```diff
@@ -97,6 +98,51 @@ class TritonAMDGPUReorderInstructionsPass
       kv.first->moveBefore(kv.second);
     opToMove.clear();
 
+    // Move writing to LDS and reading from LDS right after the loading of a
+    // tensor from global memory. There are 2 possible patterns depending on
+    // whether writing to LDS is done using an optional local_alloc argument or
+    // a local_store instruction:
+    //
+    // 1) %1 = load %ptr
+    //    %2 = local_alloc %1
+    //    %3 = local_load %2
+    //
+    // 2) %1 = load %ptr
+    //    %2 = local_alloc
+    //    %3 = local_store %1, %2
+    //    %4 = local_load %2
+    m.walk([&](ttg::LocalLoadOp op) {
+      Operation *localLoad = (Operation *)op;
+      Operation *localAlloc = localLoad->getOperand(0).getDefiningOp();
+
+      if (!localAlloc || !isa<ttg::LocalAllocOp>(localAlloc)) {
+        return;
+      }
+
+      // Case when localAlloc has no operands
+      if (localAlloc->getNumOperands() < 1) {
```

Comment:
Can we check for the # operand == 1 case first so we can do early return for that simpler one?